pkg/bootstrap/docker/host/host.go

```diff
@@ -53,22 +74,55 @@ func NewHostHelper(client *docker.Client, image, volumesDir, configDir, dataDir
 	}
 }
 
+var dockerVersion10 = semver.MustParse("1.10.0")
+
 // CanUseNsenterMounter returns true if the Docker host machine can execute findmnt through nsenter
-func (h *HostHelper) CanUseNsenterMounter() (bool, error) {
-	// For now, use a shared mount on Windows/Mac
-	// Eventually it also needs to be used on Linux, but nsenter
-	// is still needed for Docker 1.9
-	if runtime.GOOS == "darwin" || runtime.GOOS == "windows" {
+func (h *HostHelper) CanUseNsenterMounter(dockerVersion *semver.Version) (bool, error) {
+
+	// If Docker version >= 1.10 use shared volumes
+	if dockerVersion != nil && dockerVersion.GTE(dockerVersion10) {
 		return false, nil
 	}
-	rc, err := h.runner().
+	// If running on Mac or Windows do not even try Docker 1.9
+	if runtime.GOOS == "darwin" || runtime.GOOS == "windows" {
+		return false, fmt.Errorf("Docker version 1.10 or higher is required on Mac or Windows")
+	}
+	// Test that nsenter volumes can actually be used
+	err := h.EnsureHostDirectories()
+	if err != nil {
+		return false, errors.NewError("cannot verify nsenter mounts").WithCause(err).WithSolution("Install Docker version 1.10 or higher")
+	}
+	binds := []string{"/:/rootfs:ro", fmt.Sprintf("%[1]s:%[1]s", h.volumesDir)}
+	rc, err := h.runHelper.New().
+		Image(h.image).
+		DiscardContainer().
+		Privileged().
+		Bind(binds...).
+		Entrypoint("/bin/bash").
+		Command("-c", fmt.Sprintf(cmdTestNsenterMountSetup, nsEnterMount, h.volumesDir)).Run()
+	if err != nil || rc != 0 {
+		return false, errors.NewError("cannot verify nsenter mounts").WithCause(err).WithSolution("Install Docker version 1.10 or higher")
+	}
+	defer func() {
+		// cleanup
+		h.runHelper.New().Image(h.image).
+			DiscardContainer().
+			Privileged().
+			Bind(binds...).
+			Entrypoint("/bin/bash").
+			Command("-c", fmt.Sprintf(cmdTestNsenterMountCleanup, nsEnterMount, h.volumesDir)).Run()
+	}()
+	rc, err = h.runHelper.New().
 		Image(h.image).
 		DiscardContainer().
 		Privileged().
-		Bind("/:/rootfs:ro").
+		Bind(fmt.Sprintf("%[1]s/testdir:%[1]s/testdir", h.volumesDir)).
 		Entrypoint("/bin/bash").
-		Command("-c", cmdTestNsenterMount).Run()
-	return err == nil && rc == 0, nil
+		Command("-c", fmt.Sprintf(cmdTestNsenterMountVerify, h.volumesDir)).Run()
+	if err != nil || rc != 0 {
+		return false, errors.NewError("cannot use nsenter mounts.").WithCause(err).WithSolution("Install Docker version 1.10 or higher")
+	}
+	return true, nil
 }
 
 // EnsureVolumeShare ensures that the host Docker machine has a shared directory that can be used
```

Comment:
off-topic, I think it would be great if `ensureVolumesDirBindMount` and `ensureVolumesDirShare` had godoc stating what they do.